src/benchmarking.rs

```diff
@@ -17,12 +17,659 @@
+			}
+		}
+
+		T::BenchmarkHelper::set_time(HOURS);
+	}: _(RawOrigin::Signed(reporter), u8::MAX)
+
+	vote_on_multiple_disputes {
+		// Maximum votes for disputes are used in order to measure the maximum weight
+		let s in 2..T::MaxVotes::get();
+		let query_data: QueryDataOf<T> = spot_price("dot", "usd").try_into().unwrap();
+		let query_id = Keccak256::hash(query_data.as_ref()).into();
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		let another_reporter = account::<AccountIdOf<T>>("account", 2, SEED);
+		let address = Address::zero();
+		let mut votes: BoundedVec<(DisputeId, Option<bool>), T::MaxVotes> = BoundedVec::default();
+		deposit_stake::<T>(reporter.clone(), trb(1200), address)?;
+		deposit_stake::<T>(another_reporter.clone(), trb(1200), address)?;
+		T::BenchmarkHelper::set_balance(reporter.clone(), 1000);
+		T::BenchmarkHelper::set_balance(another_reporter.clone(), 1000);
+		for i in 1..=s {
+			T::BenchmarkHelper::set_time(HOURS);
+			Tellor::<T>::submit_value(RawOrigin::Signed(reporter.clone()).into(), query_id, uint_value::<T>(i * 1_000), 0, query_data.clone())?;
+
+			let timestamp = <TimeOfLastNewValue<T>>::get().unwrap();
+			let dispute_id = dispute_id(PARA_ID, query_id, timestamp);
+			if i % 2 == 0 {
+				Tellor::<T>::begin_dispute(RawOrigin::Signed(reporter.clone()).into(), query_id, timestamp, None)?;
+				let _ = votes.try_push((dispute_id, Some(true)));
+			} else {
+				Tellor::<T>::begin_dispute(RawOrigin::Signed(another_reporter.clone()).into(), query_id, timestamp, None)?;
+				let _ = votes.try_push((dispute_id, Some(false)));
+			}
+		}
+	}: _(RawOrigin::Signed(reporter), votes)
+
+	on_initialize {
+		let staking_token_price_query_data: QueryDataOf<T> =
+			spot_price("trb", "gbp").try_into().unwrap();
+		let staking_token_price_query_id = Keccak256::hash(staking_token_price_query_data.as_ref()).into();
+		let staking_to_local_token_query_data: QueryDataOf<T> =
+			spot_price("trb", "ocp").try_into().unwrap();
+		let query_data: QueryDataOf<T> = spot_price("dot", "usd").try_into().unwrap();
+		let query_id = Keccak256::hash(query_data.as_ref()).into();
+		let staking_to_local_token_query_id: QueryId =
+			Keccak256::hash(staking_to_local_token_query_data.as_ref()).into();
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		let another_reporter = account::<AccountIdOf<T>>("account", 2, SEED);
+		let user = account::<AccountIdOf<T>>("account", 3, SEED);
+		let address = Address::zero();
+		// report deposit stake
+		deposit_stake::<T>(reporter.clone(), trb(10_000), address)?;
+		deposit_stake::<T>(another_reporter.clone(), trb(1200), address)?;
+		T::BenchmarkHelper::set_balance(another_reporter.clone(), 1000);
+		T::BenchmarkHelper::set_time(HOURS);
+		// submit value
+		Tellor::<T>::submit_value(
+			RawOrigin::Signed(reporter.clone()).into(),
+			staking_token_price_query_id,
+			uint_value::<T>(50 * 10u128.pow(18)),
+			0,
+			staking_token_price_query_data.clone())?;
```

Comment:
Is it possible that the query data could be larger in this case?

Reply:
This query data will vary for each runtime, and currently we are using this in the our runtime.
That's why we are using this here, if we change the value, at the of execution, we'll get query data mismatch error.